File: analyst/src/test/java/com/zorroa/analyst/AbstractTest.java
package com.zorroa.analyst;

import com.zorroa.common.config.ApplicationProperties;
import com.zorroa.common.elastic.ElasticClientUtils;
import com.zorroa.sdk.util.FileUtils;
import org.elasticsearch.client.Client;
import org.junit.Before;
import org.junit.runner.RunWith;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.TestPropertySource;
import org.springframework.test.context.junit4.SpringRunner;
import org.springframework.test.context.web.WebAppConfiguration;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.nio.file.Path;
import java.nio.file.Paths;

/**
 * Created by chambers on 2/17/16.
 */

@RunWith(SpringRunner.class)
@SpringBootTest
@TestPropertySource("/test.properties")
@WebAppConfiguration
public abstract class AbstractTest {

    protected Logger logger = LoggerFactory.getLogger(getClass());

    @Autowired
    protected ApplicationProperties applicationProperties;

    @Autowired
    protected Client client;

    protected Path resources;

    public AbstractTest() {
        System.setProperty("zorroa.unittest", "true");
    }

    @Before
    public void __init() throws IOException, ClassNotFoundException {
        /**
         * Setup path to the test resources
         */
        resources = FileUtils.normalize(Paths.get("../../zorroa-test-data"));
    }

    public void refreshIndex() {
        ElasticClientUtils.refreshIndex(client);
    }

    public static boolean deleteRecursive(File path) throws FileNotFoundException {
        if (!path.exists()) {
            return true;
        }
        boolean ret = true;
        if (path.isDirectory()){
            for (File f : path.listFiles()){
                ret = ret && deleteRecursive(f);
            }
        }
        return ret && path.delete();
    }
}

File: analyst/src/test/kotlin/com/zorroa/analyst/AbstractTest.kt
package com.zorroa.analyst

import org.junit.Before
import org.junit.runner.RunWith
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.test.context.TestPropertySource
import org.springframework.test.context.junit4.SpringRunner
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.setup.MockMvcBuilders
import org.springframework.transaction.annotation.Transactional
import org.springframework.web.context.WebApplicationContext
import javax.sql.DataSource


@Transactional
@TestPropertySource(locations=["classpath:test.properties"])
@RunWith(SpringRunner::class)
@SpringBootTest
open abstract class AbstractTest {

    protected val logger = LoggerFactory.getLogger(javaClass)

    protected lateinit var jdbc : JdbcTemplate

    @Autowired
    fun setDatasource(dataSource: DataSource) {
        this.jdbc = JdbcTemplate(dataSource)
    }
}

@Transactional
@TestPropertySource(locations=["classpath:test.properties"])
@RunWith(SpringRunner::class)
@SpringBootTest
open abstract class AbstractMvcTest {

    protected val logger = LoggerFactory.getLogger(javaClass)

    lateinit var mockMvc: MockMvc

    @Autowired
    lateinit var wac: WebApplicationContext

    @Before
    fun setup() {
        this.mockMvc = MockMvcBuilders.webAppContextSetup(wac).build()
    }
}
